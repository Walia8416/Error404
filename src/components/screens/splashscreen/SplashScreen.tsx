import {View, Text} from 'react-native';
import React from 'react';
import {RouteStackParamList} from '../../../Routes';
import AsyncStorage from '@react-native-async-storage/async-storage';
import styles from './SplashScreens';
import ImageContainer from '../../helpers/header/ImageContainer';
import Images from '../../../constants/icon';
import {RFValue} from 'react-native-responsive-fontsize';
import {
  Bold,
  ExtraBold,
  RobMono,
  SemiBoldItalic,
} from '../../../constants/Fonts';
import {Colors} from '../../../constants/colors';
import {useAppSelector} from '../../../store/store';
import {useDispatch} from 'react-redux';
import {getStores} from '../../../store/actions/stores';
import {getorders} from '../../../store/actions/getOrders';

const SplashScreen: React.FC<RouteStackParamList<'SplashScreen'>> = ({
  navigation,
  route,
}: RouteStackParamList<'SplashScreen'>) => {
  const dispatch = useDispatch();
  React.useEffect(() => {
    setTimeout(() => {
      importData();

      getData();
    }, 50);
  });

  const importData = async () => {
    try {
      const keys = await AsyncStorage.getAllKeys();
      const result = await AsyncStorage.multiGet(keys);
      console.log(result);
      return result;
    } catch (error) {
      console.error(error);
    }
  };

  const getData = async () => {
    try {
      const dataToken = await AsyncStorage.getItem('Tokens');

      if (dataToken !== null) {
        navigation.navigate('Home');
      } else {
        navigation.navigate('SignIn');
      }
    } catch (e) {
      console.log('errors');
    }
  };

  return (
    <View style={styles.screen}>
      <View style={styles.subsection}>
        <ImageContainer name={Images.logo} style={styles.logos} />
      </View>
    </View>
  );
};
export default SplashScreen;
