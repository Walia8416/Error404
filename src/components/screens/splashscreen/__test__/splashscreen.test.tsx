import {act, create} from 'react-test-renderer';

import React from 'react';
import SplashScreen from '../SplashScreen';

const CreateTestProps = (props: Object) => ({
  navigation: {
    state: {params: {}},
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn(),
  },
  ...props,
});
let props: any;

describe('splashscreen Test', () => {
  it('splashscreen render test', () => {
    jest.useFakeTimers();
    jest.runAllTimers();
    props = CreateTestProps({});
    const splashscreen = create(
      <SplashScreen {...props} route={{params: {name: 'OnBoarding'}}} />,
    );
    expect(splashscreen).toMatchSnapshot();
  });
});
