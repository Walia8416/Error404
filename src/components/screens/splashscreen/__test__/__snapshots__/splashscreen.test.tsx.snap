// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splashscreen Test splashscreen render test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="37"
    bbWidth="236"
    fill="none"
    focusable={false}
    height="37"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 37,
          "width": 236,
        },
      ]
    }
    vbHeight={37}
    vbWidth={236}
    width="236"
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M27.11 25.9h-6.059L14.526 6.013l-.466-.463c-.466 0-.466.463-.933.463L6.602 25.9H.542c.467-1.85 1.4-4.162 2.331-7.4.932-3.7 2.33-6.938 3.263-10.175 1.398-3.7 1.864-5.55 2.33-6.012.467-.926 1.399-1.388 2.331-1.85C11.729 0 12.661 0 14.06 0c.932 0 1.864 0 2.796.463.932.462 1.865.924 2.33 1.85.467.462.933 2.312 2.331 6.012.933 2.313 1.865 6.013 3.263 10.175.932 2.313 1.864 5.087 2.33 7.4zm13.983 0H28.51V0h12.584c2.797 0 5.594 1.387 6.992 4.162 1.864 2.313 2.797 5.55 2.797 9.25 0 3.238-.933 6.013-2.797 8.325-1.398 2.776-3.729 4.163-6.992 4.163zm-.932-5.55c1.399 0 2.33-.925 3.263-2.313.932-1.387.932-3.237.932-5.55 0-1.85 0-3.237-.932-4.625-.932-1.387-1.864-2.312-3.263-2.312h-6.059v14.8h6.06z M90.966 13.875L86.771 0h-7.457l-3.73 13.412L71.39 0h-7.458L56.01 25.9h7.457l4.195-13.412L71.856 25.9h-.466 7.924l3.729-13.412L87.237 25.9 83.975 37h7.923l7.924-27.288.466-.925L102.619 0h-7.458l-2.796 9.713-1.399 4.162z M210.754 20.813c0 1.85-.466 3.237-1.865 4.162-1.398.925-2.796 1.387-4.661 1.387-1.398 0-2.796-.462-4.194-.924-1.399-.925-2.331-2.313-2.797-4.163 0 0-.466-1.85-1.398-5.087-.933-3.238-1.865-5.55-2.331-7.4-.932-2.313-1.398-3.238-2.33-3.238l-.466.463c0 .462-.467.462-.467.925v19.425h-5.593V5.55c0-1.85.932-3.237 2.331-4.162C187.915.463 189.779 0 191.644 0c2.33 0 4.661 1.387 6.059 3.7.932 1.85 1.864 4.625 2.797 8.325.932 4.162 1.864 6.937 2.33 7.862 0 .463.466.463.466.463.466.462.466.462.932.462 0 0 .467 0 .467-.462.466 0 .466-.463.466-.925V.462h5.593v20.35zm-27.034 5.55h-6.525l-6.526-19.888c0-.462-.466-.462-.466-.462-.466 0-.466 0-.466.462l-6.525 19.888h-6.526c.466-1.85 1.399-4.163 2.331-7.4 1.398-3.7 2.33-6.938 3.262-10.176 1.399-3.7 2.331-6.012 2.331-6.012.466-.925 1.398-1.387 2.33-1.85.933-.463 1.865-.463 3.263-.463.932 0 2.331 0 3.263.463.932.462 1.398.925 1.864 1.85.466 0 1.399 2.313 2.331 5.55.932 2.775 2.33 6.013 3.729 10.637.466 2.313 1.398 4.625 2.33 7.4zm41.949 0h-12.585V.463h12.585c3.263 0 5.593 1.387 7.458 4.162 1.398 2.313 2.33 5.55 2.33 8.787 0 3.7-.932 6.475-2.33 8.788-1.865 2.775-4.195 4.163-7.458 4.163zm-.932-5.55c1.398 0 2.796-.925 3.263-2.775.932-1.388 1.398-2.776 1.398-5.088 0-1.85-.466-3.7-1.398-4.625-.467-1.85-1.865-2.312-3.263-2.312h-5.593v14.8h5.593zm-68.983 6.012l-16.313-6.938v6.476h-5.594v-8.788c0-1.85.466-3.7 1.399-5.087 1.398-1.388 2.796-1.85 5.127-1.85h6.991c.932 0 1.399-.463 1.865-.925 0-.463.466-.925.466-1.388 0-.925-.466-1.387-.466-1.85-.466-.462-.933-.462-1.865-.462h-13.517V.463h14.449c2.331 0 4.195.924 5.594 2.312.932 1.387 1.398 3.238 1.398 5.55 0 1.85-.466 3.7-1.398 5.088-1.399 1.85-3.263 2.774-5.594 2.774h-3.262l10.72 4.625v6.013zm-26.102-13.412c.467.462.933 1.387 1.399 2.312 0 .925.466 1.85.466 2.775 0 1.85-.932 3.7-1.865 5.087-1.398 1.85-3.262 2.776-5.593 2.776H109.61v-5.55h13.517c.932 0 1.398-.463 1.864-.925 0-.463.467-.926.467-1.388 0-.462-.467-1.387-.467-1.387-.466-.463-.932-.925-1.864-.925H109.61v-5.55h13.517c.932 0 1.398-.463 1.864-.925 0-.463.467-.925.467-1.388 0-.925-.467-1.387-.467-1.85-.466-.462-.932-.462-1.864-.462H109.61V.463h14.449c2.331 0 4.195.924 5.593 2.312.933 1.85 1.865 3.7 1.865 5.55 0 .925-.466 1.85-.466 2.775-.466.925-.932 1.85-1.399 2.313z"
          fill={
            Array [
              1,
              "paint0_linear",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -15033699,
              0.12,
              -14976322,
              0.17,
              -14976322,
              0.3,
              -14191430,
              0.44,
              -7454051,
              0.45,
              -1767041,
              0.7,
              -1767041,
              0.8,
              -1767041,
              1,
              -2074803,
              1.1,
              -1767041,
            ]
          }
          gradientTransform={null}
          gradientUnits={1}
          name="paint0_linear"
          x1={0}
          x2={235.404}
          y1={0}
          y2="18.481"
        />
        <RNSVGClipPath
          name="clip0"
        >
          <RNSVGPath
            d="M0 0H236V37H0z"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGClipPath>
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
